client.go

```diff
@@ -275,7 +275,7 @@ func (c *Client) checksumFile(resp *Response) stateFunc {
 	if nc, err := t.copy(); err != nil {
 		resp.err = err
 		return c.closeResponse
-	} else if nc != resp.Size {
+	} else if resp.Size != -1 && nc != resp.Size {
```

Comment:
`resp.Size` is the sum of the `http.Response.ContentLength` and any bytes resumed from previous transfers. In the case that we are resuming a previous download, this is still broken. Thanks for highlighting this!